zkitter/ui
# feat: display list of users who liked a post

Closes #3 

PR chained on #39 
Depends on https://github.com/zkitter/zkitterd/pull/10

## Purpose
Reproduce an UI similar to [this](https://www.loom.com/share/8aae6dae6c1446f1800ad50bba8e1095), especially:
- [x] display number of times this post was liked between the post date and the buttons row
- [x] underline when hovering
- [x] open a modal when clicking link
- [x] show scrollable list of users in this modal

## [Demo](https://www.loom.com/share/5c40dcc58f454520bf6e1eed58e3aac0)
|||
|---|---|
|![Screenshot from 2022-10-04 09-34-25](https://user-images.githubusercontent.com/38692952/193761434-482180ee-05ee-4021-a772-ddd77889c5cf.png)|![image](https://user-images.githubusercontent.com/38692952/193761362-d9640a6a-167e-4885-a57f-19295ea932cd.png)|

## Test plan
1. Click on a post that has likes to show its details
2. Click the `X likes` link
3. A modal opens and you can see a list of users
4. Clicking outside the modal closes it
5. To test that the modal is scrollable, you can eg replace [this](https://github.com/r1oga/zkitter-ui/blob/f211d311a7b8e1a93058d70469c87009be82c14f/src/components/Post/ExpandedPost.tsx#L41) line by:
   ```
   fetchLikersByPost(messageId).then(likers => setLikers(Array(30).fill('r1oga.eth')));
   ```

## Further Comments
This PR also refactor the `Post` Component(s): it extracts them in separate files for better readability/maintainability.
